docs/markdown/snippets/swift_cxx_interoperability.md

```diff
@@ -1,13 +1,21 @@
 ## Swift/C++ interoperability is now supported
 
 It is now possible to create Swift executables that can link to C++ or
-Objective-C++ libraries. Only specifying a bridging header for the Swift
-target is required.
+Objective-C++ libraries. To enable this feature, set the target kwarg
+_swift\_interoperability\_mode_ to 'cpp'.
+
+To import C++ code, specify a bridging header in the Swift target's
+sources, or use another way such as adding a directory containing a
+Clang module map to its include path.
+
+[Note that enabling C++ interoperability is a breaking change.][1]
```

Comment:
Should this say `ABI breaking change`?

Reply:
I'm not sure if it breaks *ABI*, but it is a breaking change at compile time since consumers of a library which turns on C++ interop will also have to turn it on. That's at least what the Swift documentation says.

Reply:
That *sounds* like either ABI or API is changing, but the docs don't give enough details for me to be sure. In the docs the "breaking" seems to be related to downstream... I'm not sure what swift calls them, rust would call them a crate... needing the same option, but I don't want to confuse that with some sort of Meson change that is breaking, ie, Meson >= 1.9 needs different handling than 1.8.

Reply:
It's not an ABI change if it hides swift details and exports an unchanged C ABI, only using the flag to define how source code is bridged. Adding new C++ ABI isn't a breaking change, modifying existing ABI is.

Reply:
It could purely be a compile-time difference: library B imports some C++ code, so for swiftc to be able to load B's .swiftmodule interface, which references the C++ header and causes the compilation for library A to parse that header as well, A also has to enable C++ interop because otherwise the header is parsed in C mode, despite the final ABI being the same.

Reply:
I'd still like to clarify that what's happening is a requirement of Swift not Meson. Maybe something like:
```md
*Note*: All swift packages must be built with the same C++ interoperability
```

Reply:
How about this?

> Note: Enabling C++ interoperability in a library target is a breaking change. Swift libraries that enable it need their consumers to enable it as well, as per [the Swift documentation][1].

[1]: https://www.swift.org/documentation/cxx-interop/project-build-setup/#vending-packages-that-enable-c-interoperability